prov/rxd/src/rxd_cq.c

```diff
@@ -494,9 +494,11 @@ static struct rxd_x_entry *rxd_match_rx(struct rxd_ep *ep,
 	}
 
 	if (!match) {
+		assert(!ep->peers[base->peer].curr_unexp);
 		unexp_msg = rxd_init_unexp(ep, pkt_entry, base, op,
 					   tag, data, msg, msg_size);
-		ep->peers[base->peer].curr_rx_id = RXD_UNEXP_ID;
+		if (op)
```

Comment:
The reason for this if (op) change is not clear

Reply:
Modified this a bit in my updates to try to make it clearer but the idea is that the op header signals more data packets coming so we mark the peer as having an active unexpected message that we're buffering data into. If it doesn't have that header, it's just a single packet and we don't have to buffer any data so we can just abandon it in the dark void that is the unexpected list
☠️